tools/perf/ResultsComparer/src/ResultsComparer/Core/Reporting/MarkdownReporter.cs

```diff
@@ -0,0 +1,144 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MarkdownReporter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using MarkdownLog;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace ResultsComparer.Core.Reporting
+{
+    /// <summary>
+    /// Generates comparison reports in markdown format.
+    /// </summary>
+    public class MarkdownReporter : IReporter
+    {
+        /// <inheritdoc/>
+        public void GenerateReport(ComparerResults results, Stream destination, ComparerOptions args)
+        {
+            ComparerResult[] resultsArray = results.Results.ToArray();
+
+            if (resultsArray.Length == 0)
+            {
+                return;
+            }
+
+            using StreamWriter writer = new(destination, leaveOpen: true);
+            PrintSummary(resultsArray, writer);
+            PrintTable(resultsArray, results.MetricName, ComparisonConclusion.Worse, writer, args);
+            PrintTable(resultsArray, results.MetricName, ComparisonConclusion.Better, writer, args);
+            PrintTable(resultsArray, results.MetricName, ComparisonConclusion.New, writer, args);
+            PrintTable(resultsArray, results.MetricName, ComparisonConclusion.Missing, writer, args);
+            writer.Flush();
+        }
+
+        private static void PrintSummary(ComparerResult[] notSame, StreamWriter writer)
+
+        {
+            IEnumerable<ComparerResult> better = notSame.Where(result =>

```

Comment:
Is the logic on this line and the one that follows solid... The variable is named `better` yet the predicate is `result => result.Conclusion == ComparisonConclusion.Worse` while on the next line the variable is named `worse` yet the predicate is `result => result.Conclusion == ComparisonConclusion.Better`

Reply:
Hmm you're right.

Reply:
Actually, the tests have issues, while the tables in the markdown files are okay, the figures in the summary are mixed up (i.e. it's displaying the incorrect better and worse counts, the values should be switched).

Reply:
I've fixed it.